core/src/main/java/org/wso2/testgrid/core/util/JsonPropFileUtil.java

```diff
@@ -0,0 +1,584 @@
+                JSONObject newJsonFile = new JSONObject(newJsonFileInputs);
+                try (BufferedWriter jsonWriter = Files.newBufferedWriter(Paths.get(jsonFilePath.toString()))) {
+                    newJsonFile.write(jsonWriter);
+                    jsonWriter.write("\n");
+                } catch (Throwable ex) {
+                    logger.error("Error while persisting infra input params to " + jsonFilePath, ex);
+                }
+            }
+        } else {
+            /*
+             If input params are specified from a script save currently executing script under currentscript
+             and save the same info under script name for future use.
+             */
+            try {
+                InputStream jsonInputStream = new FileInputStream(jsonFilePath.toString());
+                JSONTokener jsonTokener = new JSONTokener(jsonInputStream);
+                JSONObject inputJson = new JSONObject(jsonTokener);
+                JSONObject generalProps = null;
+                // Contains both general and script params
+                JSONObject scriptParamsJson = new JSONObject();
+                JSONObject scriptParamsOnly = new JSONObject(properties);
+
+                if (inputJson.has("general")) {
+                    generalProps = inputJson.getJSONObject("general");
+                }
+
+                Iterator it = properties.entrySet().iterator();
+                while (it.hasNext()) {
+                    Map.Entry pair = (Map.Entry) it.next();
+                    scriptParamsJson.put((String) pair.getKey(), pair.getValue());
+                }
+
+                if (generalProps != null) {
+                    Iterator<String> genit = generalProps.keys();
+                    while (genit.hasNext()) {
+                        String key = genit.next();
+                        scriptParamsJson.put(key, generalProps.get(key));
+                    }
+                }
+
+                String scriptNameString = scriptName.get();
+                inputJson.put(scriptNameString, scriptParamsOnly);
+                inputJson.put("currentscript", scriptParamsJson);
+
+                try (BufferedWriter jsonWriter = Files.newBufferedWriter(Paths.get(jsonFilePath.toString()))) {
+                    inputJson.write(jsonWriter);
+                    jsonWriter.write("\n");
+                } catch (Throwable e) {
+                    logger.error("Error while persisting infra input params to " + jsonFilePath, e);
+                }
+
+            } catch (IOException e) {
+
+                logger.info(jsonFilePath + "created");
+                JSONObject scriptParamsJson = new JSONObject(properties);
+                JSONObject inputJson = new JSONObject();
+                String scriptNameString = scriptName.get();
+                inputJson.put(scriptNameString, scriptParamsJson);
+                inputJson.put("currentscript", scriptParamsJson);
```

Comment:
It seems you are using exceptions as a way to handle application logic. That's a code-smell.